### ISSUE TITLE
💅 lint/correctness/noConstantCondition does not consider enums when apply the rule

### ISSUE THREAD

**Issue by jpike88** [biomejs/biome#2965]
### Environment information

```bash
enum Enum {
Blah = 3
}

if(Enum.Blah){ 

}
```

### Rule name

lint/correctness/noConstantCondition

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=ZQBuAHUAbQAgAE0AeQBFAG4AdQBtACAAewAKACAAQQAgAD0AIAAzAAoAfQAKAAoAaQBmACgATQB5AEUAbgB1AG0ALgBBACkAewAKACAAIAAvAC8AIABiAGEAZAAKAH0A

### Expected result

enums are constants and should be treated as such in this rule

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by jpike88** [comment]
My god... I didn't know that was possible.

It should be noted though that computed values are only allowed in numeric enums. And once computed they are fixed for the duration of runtime. So in that sense they behave as a constant and having a truthy if statement like that is almost always going to be an accident

**Comment by ah-yu** [comment]
> And once computed they are fixed for the duration of runtime

No, they are not fixed.

```bash
zheyzhang@HANM18365 react % cat foo.ts 
enum Foo {
    A,
    B = Math.random(),
}
console.log(Foo.B);
zheyzhang@HANM18365 react % deno run foo.ts 
0.2797788528053844
zheyzhang@HANM18365 react % deno run foo.ts
0.15004983018100404
zheyzhang@HANM18365 react % deno run foo.ts
0.8717746009202871
zheyzhang@HANM18365 react % 
```

**Comment by jpike88** [comment]
'for the duration of runtime' so in that sense they are fixed and behave like a constant as they are a readonly property.

<img width="591" alt="Screenshot 2024-05-24 at 3 40 38 PM" src="https://github.com/biomejs/biome/assets/9585787/22a7f3e4-3c74-41ad-a819-143d0ed916c4">

**Comment by ah-yu** [comment]
Sorry, I misunderstood.

While the value is determined at runtime, we cannot know it in advance. If the computed value is `0`, it will be treated as false in an if statement.

**Comment by Conaclos** [comment]
This is out of scope of the current rule. This requires type info.
